sormas-backend/src/main/resources/sql/sormas_schema.sql

```diff
@@ -12958,4 +12958,47 @@ $$ LANGUAGE plpgsql;
 
 INSERT INTO schema_version (version_number, comment) VALUES (541, 'Update environment sample deletion dependencies #12887');
 
+-- 2024-02-09 Remove entity specific PERFORM_BULK_OPERATIONS user rights #10994
+DO $$
+  DECLARE
+     rec RECORD;
+	 count_perform_bulk_actions_right integer;
+  BEGIN
+       FOR rec IN SELECT id FROM userroles
+           LOOP
+               count_perform_bulk_actions_right = (SELECT count(*) FROM userroles_userrights WHERE userrole_id = rec.id and userright = 'PERFORM_BULK_OPERATIONS');
+               IF ((SELECT exists(SELECT userrole_id FROM userroles_userrights where userrole_id = rec.id and userright = 'PERFORM_BULK_OPERATIONS_CASE_SAMPLES')) = true) THEN
```

Comment:
I believe this could be simplified by moving the `IF count_perform_bulk_actions_right` part to the top of the loop (you wouldn't even need the variable anymore because you can just perform the IF on the query) and then only execute the rest of the code if the condition is matched; you can then move the DELETE calls to the bottom of the loop after closing the IF. Would get rid of redundant code and make it easier to understand.